File: src/main/java/com/sothawo/springdataelastictest/NetworkRepository.java
/*
 * (c) Copyright 2019 sothawo
 */
package com.sothawo.springdataelastictest;

import java.util.List;

import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
public interface NetworkRepository extends ElasticsearchRepository<Network, String> {
	List<Network> findByCidr(String ip);
}

File: src/main/kotlin/com/sothawo/springdataelastictest/NetworkRepository.kt
/*
 * (c) Copyright 2019 sothawo
 */
package com.sothawo.springdataelastictest

import org.springframework.data.elasticsearch.repository.ElasticsearchRepository

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
interface NetworkRepository : ElasticsearchRepository<Network, String> {
    fun findByCidr(ip: String): List<Network>
}
